### DISCUSSION TITLE
feature: shorthand chained comparisons

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#5308] [Ideas]
Add support for shorthand comparisons like `x < y < z <= p < q` (consistent direction, possibly differing strictness). An expanded form could also allow `x < a, b, c < y`.

**Comment by mlutze** [comment]
How does it work with effects and short circuiting?

**Comment by magnus-madsen** [comment]
This idea was discussed before.

There is also an issue with `false <= true` is true which can leads to all kinds of puzzlers :)

**Comment by JonathanStarup** [comment]
> There is also an issue with false <= true is true

Oh yeah, that sounds ambiguous.

> How does it work with effects and short circuiting?

I think bout eager and short circuited evaluation could make sense.
